libs/s25main/figures/nofPassiveWorker.cpp

```diff
@@ -32,7 +35,29 @@ void nofPassiveWorker::AbrogateWorkplace() {}
 /// Zeichnen
 void nofPassiveWorker::Draw(DrawPoint drawPt)
 {
-    DrawWalking(drawPt);
+    switch(job_)
+    {
+        case Job::PackDonkey:
+        {
+            const unsigned ani_step = CalcWalkAnimationFrame();
+            drawPt = InterpolateWalkDrawPos(drawPt);
+
+            LOADER.GetMapTexture(2000 + rttr::enum_cast(GetCurMoveDir() + 3u) * 8 + ani_step)->DrawFull(drawPt);
+            LOADER.GetMapTexture(2048 + rttr::enum_cast(GetCurMoveDir()) % 3)->DrawFull(drawPt, COLOR_SHADOW);
+        }
+        break;
+        case Job::CharBurner: DrawWalking(drawPt, "charburner_bobs", 53); break;
+        case Job::Vintner:
+            DrawWalking(drawPt, "wine_bobs", wineaddon::bobIndex[wineaddon::BobTypes::VINTNER_WALKING]);
+            break;
+        case Job::Winegrower:
+            DrawWalking(drawPt, "wine_bobs", wineaddon::bobIndex[wineaddon::BobTypes::WINEGROWER_WALKING_WITH_SHOVEL]);
+            break;
+        case Job::TempleServant:
+            DrawWalking(drawPt, "wine_bobs", wineaddon::bobIndex[wineaddon::BobTypes::TEMPLESERVANT_WALKING]);
+            break;
+        default: DrawWalkingBobJobs(drawPt, job_); break;
+    }
```

Comment:
Besides the new 3 cases, this is a copy of the code in `nofFigure::DrawWalking`. Why this change?

Reply:
Ok duplicating code is never a good idea especially as we now have here exactly the code we wanted to avoid to have in `noFigure::DrawWalking`   
So yes we need to move this code into `DrawWalking` and make it non-virtual again :-/

I would go so far and replace the `DrawWalkingBobJobs(drawPt, job_)` together with the comment in nofActive/PassiveSoldier by `DrawWalking`.   
Maybe even all other calls to `DrawWalkingBobJobs`, except for the one in nofScout where the job argument is not `job_`, as `DrawWalking(drawPos)` is cleaner than e.g. `DrawWalkingBobJobs(drawPt, Job::Geologist);` removing the question whether anything special is done there.

What do you think? Relevant for this PR is only the de-duplication. If you think it is worth it or have a better idea check that 2nd part.

Reply:
Reverted drawwalking virtual and remove code from nofPassiveWorker

Reply:
Will not do the second part now